dhcpv4/server_test.go

```diff
@@ -0,0 +1,145 @@
+// +build integration
+
+package dhcpv4
+
+import (
+	"errors"
+	"log"
+	"math/rand"
+	"net"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/require"
+)
+
+func init() {
+	// initialize seed. This is generally bad, but "good enough"
+	// to generate random ports for these tests
+	rand.Seed(time.Now().UTC().UnixNano())
+}
+
+func randPort() int {
+	// can't use port 0 with raw sockets, so until we implement
+	// a non-raw-sockets client for non-static ports, we have to
+	// deal with this "randomness"
+	return 1024 + rand.Intn(65536-1024)
+}
+
+// DORAHandler is a server handler suitable for DORA transactions
+func DORAHandler(conn net.PacketConn, peer net.Addr, m *DHCPv4) {
+	if m == nil {
+		log.Printf("Packet is nil!")
+		return
+	}
+	if m.Opcode() != OpcodeBootRequest {
+		log.Printf("Not a BootRequest!")
+		return
+	}
+	response, err := New()
+	if err != nil {
+		log.Printf("New DHCPv4 packet creation failed: %v", err)
+		return
+	}
+	// TODO implement NewOfferFromDiscovery and NewAckFromRequest,
```

Comment:
How about using `NewReplyFromRequest` here?

Reply:
you're damn right, sir!